db.go

```diff
@@ -64,26 +65,57 @@ CREATE TABLE %s (
     total_uptime TEXT,
     total_downtime TEXT
 );`
+
+	// %s will be replaced by the table name
+	statSaveSchema = `INSERT INTO %s (
+	event_type,
+	timestamp,
+	addr,
+	hostname,
+	port,
+	hostname_resolve_retries,
+	total_successful_probes,
+	total_unsuccessful_probes,
+	never_succeed_probe,
+	never_failed_probe,
+	last_successful_probe,
+	last_unsuccessful_probe,
+	total_packets,
+	total_packet_loss,
+	total_uptime,
+	total_downtime,
+	longest_uptime,
+	longest_uptime_start,
+	longest_uptime_end,
+	longest_downtime,
+	longest_downtime_start,
+	longest_downtime_end,
+	latency_min,
+	latency_avg,
+	latency_max,
+	start_time,
+	end_time,
+	total_duration) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);`
 )
 
 // newDb creates a newDb with the given path and returns `database` struct
-func newDb(args []string, dbPath string) database {
+func newDb(args []string, dbPath string) *database {
 	tableName := newTableName(args)
 	tableSchema := fmt.Sprintf(tableSchema, tableName)
 
-	db, err := sql.Open("sqlite3", dbPath)
+	conn, err := sqlite.OpenConn(dbPath, sqlite.OpenCreate, sqlite.OpenReadWrite)
 	if err != nil {
 		colorRed("\nError while creating the database %q: %s\n", dbPath, err)
 		os.Exit(1)
 	}
 
-	_, err = db.Exec(tableSchema)
+	err = sqlitex.Execute(conn, tableSchema, &sqlitex.ExecOptions{})
 	if err != nil {
+		// 	// TODO: add better err messege
```

Comment:
Do we need this TODO?

Reply:
if you think the error message is not stable then please subject a better error message.

Reply:
I think it is fine for now